src/bvals/bvals_var.cpp

```diff
@@ -112,7 +112,13 @@ void BoundaryVariable::CopyVariableBufferSameProcess(NeighborBlock &nb, int ssiz
   MeshBlock *ptarget_block = pmy_mesh_->FindMeshBlock(nb.snb.gid);
   // 2) which element in vector of BoundaryVariable *?
   BoundaryData<> *ptarget_bdata = &(ptarget_block->pbval->bvars[bvar_index]->bd_var_);
-  ptarget_block->deep_copy(ptarget_bdata->recv[nb.targetid], bd_var_.send[nb.bufid]);
+  // TODO(pgrete) ensure/double check that the data on target has already been used before
+  // overwriting it
+  // using source block for deep copy to ensure physics kernels are done
+  MeshBlock *psource_block = pmy_block_;
+  psource_block->deep_copy(ptarget_bdata->recv[nb.targetid], bd_var_.send[nb.bufid]);
+  // need fence here as deep_copy is async
+  psource_block->exec_space.fence();
```

Comment:
For consistency sake with `par_for` and `deep_copy`, could you add a `Fence` method to the `MeshBlock`?

Reply:
Yes, good point. Will do.